go/vt/sqlannotations/sqlannotations.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package sqlannotations provides functions
+// for annotating DML statements with keyspace-id
+// comments and parsing them. These annotations
+// are used during filtered-replication to route
+// the DML statement to the correct shard.
+package sqlannotations
+
+import (
+	"encoding/hex"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// AnnotateKeyspaceIdIfDML annotates 'sql' with the given keyspace id
+// if 'sql' is a DML statement; otherwise, it does nothing.
+func AnnotateKeyspaceIdIfDML(keyspaceId []byte, sql *string) {
```

Comment:
I personally prefer a pointer since it's less verbose and I feel like in most call sites the "&" and function name would indicate that the parameter  is changed; however, it's not critical so I changed it to use a return value instead.

I do like the "IfDML syntax since it makes it more explicit to the reader what is actually going
on without having to look in the function comment or code. So I'm leaving that in. Let me know if you want to discuss this further

So I'd like to have:
func AddKeyspaceIDIfDML(sql string, keyspaceID [] byte) string

As per @sougou's comment above, I also have:
func AddIfDML(sql string, keyspaceIDs [][]byte) string
that adds the right comment depending on keyspaceIDs.